src/api/entities/SecurityToken/TokenHolders.ts

```diff
@@ -27,16 +42,56 @@ export class TokenHolders extends Namespace<SecurityToken> {
 
     const { entries, lastKey: next } = await requestPaginated(query.asset.balanceOf, {
       arg: rawTicker,
-      paginationOpts,
+      paginationOpts: opts?.pagination || undefined,
     });
 
-    const data: IdentityBalance[] = entries.map(([storageKey, balance]) => ({
-      identity: new Identity(
-        { did: identityIdToString(storageKey.args[1] as IdentityId) },
-        context
-      ),
-      balance: balanceToBigNumber(balance),
-    }));
+    let data: IdentityBalance[] = [];
+
+    const securityToken = new SecurityToken({ ticker }, context);
+
+    const areFrozen = opts?.mintStatus ? await securityToken.transfers.areFrozen() : undefined;
+
+    if (opts?.mintStatus && !areFrozen) {
+      const entriesChunks = chunk(entries, 10);
```

Comment:
There is a similar statement with a hardcoded 10 in `procedures/issueTokens.ts`. Please refactor both numbers into a `MAX_CONCURRENT_REQUESTS` variable in `constants.ts` and change it to a bigger number (something like 200 will do for now)